packages/code-studio/src/main/AppMainContainer.tsx

```diff
@@ -49,44 +70,119 @@ import {
   setActiveTool as setActiveToolAction,
   updateWorkspaceData as updateWorkspaceDataAction,
   getPlugins,
+  Workspace,
+  DeephavenPluginModuleMap,
+  WorkspaceData,
+  RootState,
 } from '@deephaven/redux';
 import { PromiseUtils } from '@deephaven/utils';
+import GoldenLayout, { ItemConfigType } from '@deephaven/golden-layout';
 import JSZip from 'jszip';
 import SettingsMenu from '../settings/SettingsMenu';
 import AppControlsMenu from './AppControlsMenu';
 import { getLayoutStorage } from '../redux';
 import Logo from '../settings/community-wordmark-app.svg';
 import './AppMainContainer.scss';
-import WidgetList from './WidgetList';
-import { createChartModel, createGridModel } from './WidgetUtils';
+import WidgetList, { WindowMouseEvent } from './WidgetList';
+import {
+  createChartModel,
+  createGridModel,
+  GridPanelMetadata,
+} from './WidgetUtils';
 import EmptyDashboard from './EmptyDashboard';
 import UserLayoutUtils from './UserLayoutUtils';
 import DownloadServiceWorkerUtils from '../DownloadServiceWorkerUtils';
 import { PluginUtils } from '../plugins';
+import LayoutStorage from '../storage/LayoutStorage';
 
 const log = Log.module('AppMainContainer');
 
-export class AppMainContainer extends Component {
-  static handleWindowBeforeUnload(event) {
+type InputFileFormat =
+  | string
+  | number[]
+  | Uint8Array
+  | ArrayBuffer
+  | Blob
+  | NodeJS.ReadableStream;
+
+export type DashboadData = {
```

Comment:
I was so confused why this wasn't conflicting with the `DashboardData` that already exists in redux - it's because it's misspelled `DashboadData` (missing `r`).

I think just name this `AppDashboardData` instead to be clear.